### PR TITLE
Use the Vue store for spaces

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
Using the store for spaces integrates them seamlessly in our ecosystem and makes it easier to develop spaces even further. E.g. the properties of a space can now be altered without fetching all spaces again. This was achieved by introducing a "buildSpace" method, that transforms a space into a more generic resource object (just like regular files or shares).

## Motivation and Context
Groundwork for https://github.com/owncloud/web/issues/6284 (and basically all space-related issues)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingAndUpload</strong> https://drone.owncloud.com/owncloud/web/22694/66/1
The following scenarios passed on retry:
- webUIUpload/upload.feature:94

File: packages/web-app-files/src/helpers/resources.js [hunk 1]
```diff
@@ -102,6 +102,80 @@ export function buildResource(resource) {
...
+
+    if (space.root.deleted) {
+      disabled = space.root?.deleted?.state === 'trashed'
+    }
+  }
+
+  return {
+    id: space.id,
+    fileId: '',
+    mimeType: '',
```
Comment (JanAckermann) [inline packages/web-app-files/src/helpers/resources.js:125]: Do we need those fields, afaik they will never be set

File: packages/web-app-files/src/views/spaces/Project.vue [hunk 2]
```diff
@@ -193,23 +204,22 @@ export default {
...
-      loadReadmeTask.perform(ref)
-      loadImageTask.perform(ref)
-      loadFilesListTask.perform(ref, sameRoute, path)
+      yield loadFilesListTask.perform(ref, sameRoute, path)
+
+      // Only load when in space root, no need to fetch in subdirectories
+      if (!path) {
+        loadReadmeTask.perform(ref)
+        loadImageTask.perform(ref)
+      }
```
Comment (JanAckermann) [inline packages/web-app-files/src/views/spaces/Project.vue:218]: :+1:

**Review by JanAckermann — APPROVED** [review]
Looks Impressive to me

File: packages/web-app-files/src/helpers/resources.js (hunk 1 above)
Comment (kulmann) [reply packages/web-app-files/src/helpers/resources.js:125]: Depends on how the respective fields are used throughout the code. Sometimes there are string modifications applied - which don't work on `undefined`. IMO ok to keep them defined and empty, like it's done here. Will be cleaned up anyway once we refactor this to TypeScript.

File: packages/web-app-files/src/helpers/resources.js [hunk 3]
```diff
@@ -102,6 +102,80 @@ export function buildResource(resource) {
...
+  if (space.special) {
+    spaceImageData = space.special.find((el) => el.specialFolder.name === 'image')
+    spaceReadmeData = space.special.find((el) => el.specialFolder.name === 'readme')
+  }
+
+  if (space.root) {
+    spacePermissions = space.root.permissions
+
+    if (space.root.deleted) {
+      disabled = space.root?.deleted?.state === 'trashed'
```
Comment (kulmann) [inline packages/web-app-files/src/helpers/resources.js:118]:
```suggestion
      disabled = space.root.deleted?.state === 'trashed'
```

File: packages/web-app-files/src/mixins/spaces/actions/disable.js [hunk 4]
```diff
@@ -62,7 +63,11 @@ export default {
         .deleteDrive(id)
         .then(() => {
           this.hideModal()
-          this.loadSpacesTask.perform(this)
+          this.UPDATE_RESOURCE_FIELD({
+            id: id,
```
Comment (kulmann) [inline packages/web-app-files/src/mixins/spaces/actions/disable.js:67]:
```suggestion
            id,
```

File: packages/web-app-files/src/mixins/spaces/actions/editDescription.js [hunk 5]
```diff
@@ -57,7 +58,11 @@ export default {
         .updateDrive(id, { description }, {})
         .then(() => {
           this.hideModal()
-          this.loadSpacesTask.perform(this)
+          this.UPDATE_RESOURCE_FIELD({
+            id: id,
```
Comment (kulmann) [inline packages/web-app-files/src/mixins/spaces/actions/editDescription.js:62]:
```suggestion
            id,
```

File: packages/web-app-files/src/mixins/spaces/actions/rename.js [hunk 6]
```diff
@@ -64,7 +65,11 @@ export default {
         .updateDrive(id, { name }, {})
         .then(() => {
           this.hideModal()
-          this.loadSpacesTask.perform(this)
+          this.UPDATE_RESOURCE_FIELD({
+            id: id,
```
Comment (kulmann) [inline packages/web-app-files/src/mixins/spaces/actions/rename.js:69]:
```suggestion
            id,
```

File: packages/web-app-files/src/mixins/spaces/actions/restore.js [hunk 7]
```diff
@@ -70,7 +71,11 @@ export default {
         )
         .then(() => {
           this.hideModal()
-          this.loadSpacesTask.perform(this)
+          this.UPDATE_RESOURCE_FIELD({
+            id: id,
```
Comment (kulmann) [inline packages/web-app-files/src/mixins/spaces/actions/restore.js:75]:
```suggestion
            id,
```

File: packages/web-app-files/src/helpers/resources.js (hunk 1 above)
Comment (JammingBen) [reply packages/web-app-files/src/helpers/resources.js:125]: The way I understood @kulmann , these fields should be available to keep they resource object "aligned" with the other resources. Maybe I got it wrong though, @kulmann ? Jan is right, some of these fields will never be set.

**Review by kulmann — CHANGES_REQUESTED** [review]

**Review by kulmann — APPROVED** [review]
🚀

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublic1</strong> https://drone.owncloud.com/owncloud/web/22718/67/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISTrashbinUploadMoveJourney</strong> https://drone.owncloud.com/owncloud/web/22721/69/1
The following scenarios passed on retry:
- webUIUserJourney/journey1.feature:11

**Pull request merged**